src/Npgsql/ClusterStateCache.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Npgsql.Util;
+using System;
+using System.Collections.Concurrent;
+
+namespace Npgsql
+{
+    static class ClusterStateCache
+    {
+        static readonly ConcurrentDictionary<ClusterIdentifier, ClusterInfo> Clusters = new();
+
+        static readonly TimeSpan ClusterStateExpiration = TimeSpan.FromSeconds(10); // TODO: add a param?
```

Comment:
I think I'd definitely want control over that from the consuming application

Reply:
Done